### DISCUSSION TITLE
monitor-any-sql-metrics-with-netdata/

### DISCUSSION THREAD

**Discussion by giscus** [netdata/blog#197] [General]
# monitor-any-sql-metrics-with-netdata/

Using the Netdata Pandas collector to monitor and wrangle any SQL metrics you want.

https://blog.netdata.cloud/monitor-any-sql-metrics-with-netdata/

**Comment by Theo-Dancoisne** [comment]
@andrewm4894 Hello !
I just want to report an issue in this blog post:

I'm on netdata v1.38.1 and when i try this, python.d.plugin in debug mode says that `create_engine` is not defined.
![image](https://user-images.githubusercontent.com/105285931/227608752-730959c5-22a6-45f9-b931-dbfcdf417497.png)

You had to add `sqlalchemy import create_engine` in `/usr/libexec/netdata/python.d/pandas.chart.py`.

Have a nice weekend.

**Reply by andrewm4894** [reply]
Oh thanks, I will update to blog post to reflect this, or maybe see if I can update the collector to try and do any sqlalchemy imports. 

I think it might work if do sqlalchemy.create_engine()

**Reply by andrewm4894** [reply]
Oh actually maybe it's a version thing as I think the collector tries to import it. 

https://github.com/netdata/netdata/blob/03dc41202ac4d65ca156095f317580d71e8603f3/collectors/python.d.plugin/pandas/pandas.chart.py#L16

**Reply by andrewm4894** [reply]
I think could be that earlier netdata versions did not have this. Thanks for letting us know!

You have a good weekend too!

**Reply by Theo-Dancoisne** [reply]
Thank you,

Actually I'm on the very last stable release (I installed netdata yesterday) and I don't have this lines https://github.com/netdata/netdata/blob/master/collectors/python.d.plugin/pandas/pandas.chart.py#L16

Will it also be more consistent and easier for people to create a python collector for this ?

**Reply by andrewm4894** [reply]
I believe there will be a new stable release in a few weeks which will include these two PR's that added the `read_sql()` functionality and fixed one issue with self.warning(). 

https://github.com/netdata/netdata/pull/14563
https://github.com/netdata/netdata/pull/14736

So I'd say best option in general would be to wait for that or use latest nightly which would have it.

Alternative would be, like you mention above, just update `/usr/libexec/netdata/python.d/pandas.chart.py` manually to be same as the [latest version](https://raw.githubusercontent.com/netdata/netdata/master/collectors/python.d.plugin/pandas/pandas.chart.py) of that file, but that might be a bit annoying or manual for many.
